### DISCUSSION TITLE
add doc "how to test own fixer"

### DISCUSSION THREAD

**Discussion by dozer111** [PHP-CS-Fixer/PHP-CS-Fixer#8137] [Q&A]
I have my own private repo with fixers.

Now - I want to add tests on them.

I try to find some info in your site and git repo "how to test my own fixer", and find nothing

If it is exists - where can i find it?

If it is not - it would be very helpful to create doc about it.

For example, rector has such doc - https://getrector.com/documentation/writing-tests-for-custom-rule

**Comment by Wirone** [comment]
You may look at how e.g. @kubawerlos did it in his [custom fixers](https://github.com/kubawerlos/php-cs-fixer-custom-fixers/tree/main/tests), but also follow the tests in the main repository. I am not sure if we need documentation for that 🤔.

**Reply by dozer111** [reply]
why not?

The fixer creation ideally contain both parts: create a fixer and create test for it to be always sure that fixer works as expected

You have doc "how to create own fixer". It would be a huge DX improvement if this doc also have block "how to write test on custom fixer", because I am(as your code user) no longer need to find this information chaotically. All the required knowledge in one place.

Additional plus - you, as a maintainer know a lot about "how to". So you can write how to do it "the best"
